### PR TITLE
Non safepointing text render

### PR DESCRIPTION
**Pull request by TimWSpence**
`StringWriter` is based on `java.io.StringBuffer`, which synchronizes every write. Our usage of `Writer` instances is entirely local so this is a _huge_ safepoint penalty to pay for no benefit.

### CODE REVIEW

**Comment by TimWSpence** [comment]
There's still a `println` in the test - I was getting some really odd compilation issues locally so just wanted to push this to see what CI thinks

File: prometheus-metrics/src/test/scala/org/http4s/metrics/prometheus/PrometheusExportServiceSuite.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.http4s.dsl.io._
+import org.http4s.implicits._
+
+class PrometheusClientMetricsSuite extends CatsEffectSuite {
+
+  test("Returns Prometheus-format") {
+    PrometheusExportService.build[IO].use { svc =>
+      svc.routes.orNotFound
+        .run(GET(uri"/metrics"))
+        .flatMap(resp => resp.body.compile.to(Array).map(new String(_)))
```
Comment (hamnis) [inline prometheus-metrics/src/test/scala/org/http4s/metrics/prometheus/PrometheusExportServiceSuite.scala:31]: rather use `resp.as[String]`

Comment (TimWSpence) [reply prometheus-metrics/src/test/scala/org/http4s/metrics/prometheus/PrometheusExportServiceSuite.scala:31]: Yes that is much better 😅

**Comment by hamnis** [comment]
Looks good. Maybe squash the commits into one?

**Comment by TimWSpence** [comment]
> Looks good. Maybe squash the commits into one?

Thanks! I kept the first few commits as they vaguely made sense on their own but squashed all the cross-compilation / build pain at the end. Or shall I squash them all?

**Review by hamnis — APPROVED** [review]
Thanks.

**Comment by hamnis** [comment]
Released as [0.24.2](https://github.com/http4s/http4s-prometheus-metrics/tree/v0.24.2)

**Pull request merged**